src/components/tinyangel/TinyAngel.tsx

```diff
@@ -0,0 +1,244 @@
+            setRewardState({ validatorAddresses })
+        })();
+    })
+
+    useEffect(() => {
+        if ( !user_address || ustSwapRateMap === undefined ) return;
+
+        const getTinyBalances = setTimeout(tinyBalanceSetter, 200)
+        return () => clearTimeout(getTinyBalances);
+    }, [ user_address, nativeWalletState.tinyValue, ustSwapRateMap ]);
+
+    useEffect(() => {
+        if ( !user_address ) return;
+
+        const getTotalRewards = setTimeout(rewardStateSetter, 200);
+        return () => clearTimeout(getTotalRewards);
+    }, [ user_address, rewardState.tinyReward ])
+
+    return (
+        <>
+        <Container sx={{maxWidth: '1200px', padding: '10px'}}>
+            <Grid container spacing={2}>
+                <Grid item xs={12} sm={6}>
+                    <Paper elevation={3} style={{height: 'fit-content'}}>
+                        <Stack spacing={1} sx={{padding: '30px'}}>
+                            <Typography 
+                            variant="h5" 
+                            width="100%" 
+                            textAlign="center">
+                                Donate Tiny Wallet Balance
+                            </Typography>
+
+                            <Stack
+                            padding="15px 0">
+                                <Typography>
+                                    Tiny Balance Upper Limit (in UST)
+                                </Typography>
+                                <Slider 
+                                onChange={(e: any) => setNativeWalletState({ tinyValue: +e.target.value })}
+                                color="primary"
+                                min={0.5} 
+                                max={5}
+                                step={0.1}
+                                valueLabelDisplay="auto"/>
+                            </Stack>
+
+                            { nativeWalletState.tinyBalances.length === 0 &&
+                                <Typography>No Tiny Balances</Typography>
+                            }
+
+                            {nativeWalletState.tinyBalances.length > 0 && 
+                                <Grid 
+                                container 
+                                gap="10px">
+                                    {nativeWalletState.tinyBalances.map(balance => (
+                                        <Grid
+                                        border="1px solid white"
+                                        borderRadius="10px">
+                                            <Typography variant="h6" sx={{margin: '10px'}}>
+                                                {visualDenomName.get(balance.denom)}
+                                            </Typography>
+                                            <Typography variant="h6" sx={{margin: '10px'}}>
+                                                {toTerraAmount(balance.amount).toFixed(6)}
```

Comment:
Let's add the UST value here as well to give users the peace of mind that their values truly are under threshold